modules/packages/Search.chpl

```diff
@@ -157,7 +167,7 @@ proc linearSearch(Data:[?Dom], val, comparator:?rec=defaultComparator, lo=Dom.al
    :rtype: (`bool`, `Dom.idxType`)
 
  */
-proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom.alignedLow, in hi=Dom.alignedHigh) {
+proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom.alignedLow, in hi=Dom.alignedHigh, param bound : Bound = Bound.ANY) {
```

Comment:
I've said this several times on the original issue, but I don't see a strong motivation for making this argument a `param` (which will result in copies of the routine getting stamped out for each possible value and prevents a user from making the choice dynamically).  Do you think there's an argument for doing so?  (e.g., will it result in significant increases in execution time that warrants the restricted interface and increase in generated code size?)

Reply:
No, I dont have any particular argument for it. Whats your suggestion? Should I replace `param` with `const` ? Or just let be a variable altogether?

Reply:
I'd leave the argument intent blank (which causes it to be `const in` for enums).

Reply:
Made the respective changes :)